### PR TITLE
Add base builder images

### PR DESCRIPTION
**Pull request by danielingrey**
So the purpose of this repo is to eventually replace the docker builder images from infrastructure. I've added the E2E tester image and it would be good if this repo could be a place for all and any images we need during CI.

Small changes:
* we no longer compile node
* the images are tagged with the latest node version (we'll have to manually increase the version number for npm)
* no centOS images as there is no official distro. We may want to move the bat over to debian or alpine if possible

### CODE REVIEW

**Comment by jonnyarnold** [comment]
I can't remember for certain why we had to go to CentOS, but I know we tried Alpine and (I think) Debian and they didn't work - YMMV. I think it's either the font rendering, the ImageMagick image manipulation, or maybe both.

**Comment by danielingrey** [comment]
worth reviewing again for sure though. I don't see why at the very least debian wouldn't work. I'd say that's a separate piece of work though

File: build-images/alpine-node-10/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+RUN apk add --no-cache --update \
+  build-base \
+  python \
+  automake \
+  autoconf \
+  openjdk8-jre \
+  libtool \
+  libpng-dev \
+  protobuf \
+  unzip
```
[the remark this answers is not in the document]
Comment (danielingrey) [reply build-images/alpine-node-10/Dockerfile]: That's a good shout. I'll draw something up and add it to the readme

[the remark this answers is not in the document]
Comment (danielingrey) [reply build-images/alpine-node-10/Dockerfile]: I've added a list of what these are used by into the readme https://github.com/trussle/docker-builder/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R4

**Pull request merged**